src/Razor/src/Microsoft.VisualStudio.RazorExtension.Dependencies/Microsoft.VisualStudio.RazorExtension.Dependencies.csproj

```diff
@@ -43,13 +43,37 @@
+    <!-- Reference the Roslyn dependencies so that Preview builds work -->
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Features" Version="$(Tooling_MicrosoftCodeAnalysisCSharpFeaturesPackageVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.EditorFeatures" Version="$(RoslynPackageVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.EditorFeatures.Wpf" Version="$(RoslynPackageVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.ExternalAccess.Razor" Version="$(Tooling_MicrosoftCodeAnalysisExternalAccessRazorPackageVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.ExternalAccess.FSharp" Version="$(Tooling_MicrosoftCodeAnalysisExternalAccessRazorPackageVersion)" />
+    <PackageReference Include="Microsoft.VisualStudio.LanguageServices" Version="$(Tooling_MicrosoftVisualStudioLanguageServicesPackageVersion)" />
+
     <!-- Reference the LSP protocol dlls so that we can include them in the code base and output them with the VSIX. -->
     <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Protocol" Version="$(MicrosoftVisualStudioLanguageServerProtocolPackageVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Protocol.Extensions" Version="$(MicrosoftVisualStudioLanguageServerProtocolExtensionsPackageVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Protocol.Internal" Version="$(MicrosoftVisualStudioLanguageServerProtocolInternalPackageVersion)" />
   </ItemGroup>
 
   <ItemGroup>
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.CSharp.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.CSharp.Features.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.CSharp.Workspaces.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.EditorFeatures.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.EditorFeatures.Text.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.EditorFeatures.Wpf.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.ExternalAccess.FSharp.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.ExternalAccess.Razor.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.InteractiveHost.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.Features.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.LanguageServer.Protocol.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.Remote.Workspaces.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.Workspaces.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.CodeAnalysis.VisualBasic.Workspaces.dll" />
+    <VSIXSourceItem Include="$(OutputPath)Microsoft.VisualStudio.LanguageServices.dll" />
```

Comment:
Why does all of this need to be in our VSIX?

Reply:
More to the point, if these all get into the VSIX we insert into VS, we can expect RPS failures and Dave Kean to come knocking :D

If this is only used for our integration tests, and is here to allow us to bring an older VS instance up to our requirements then I think it should be okay.

Reply:
I think this is just for integration tests. However, I'm also curious how this particular set of references was chosen. For instance, why would we include Microsoft.CodeAnalysis.VisualBasic.Workspaces.dll, but not Microsoft.CodeAnalysis.VisualBasic.dll?

Reply:
This stuff is NOT going into our (shipping) VSIX. This project was created to handle point-in-time issues with the fact that we are shipping into VSMain but can only test against Public Preview for our CI tests. Essentially we can use this to binding-redirect things to the version we're looking for when running the Experimental instance or Integration tests, but we do not ship it.

Reply:
(Realized I forgot to answer how the references were chosen).

I arrived at this list by seeing what assemblies showed up in the MEF error list after updating Roslyn and running this on VSPreview. After your comments I trimmed that down to "Those assemblies which show up in the MEF error list which seem to actually cause problems for the Razor scenarios". So all the VB and F# stuff is out.